lib/fuse.c

```diff
@@ -3673,6 +3676,18 @@ static int readdir_fill_from_list(fuse_req_t req, struct fuse_dh *dh,
 				.ino = 0,
 				.attr = de->stat,
 			};
+
+			if (flags & FUSE_READDIR_PLUS) {
```

Comment:
I need to go over all details of this patch and readdir-plus code, but here you are already in the condition `if (flags & FUSE_READDIR_PLUS)` - no need to check another times.

Reply:
thanks, fixed.
Please take your time, i'm obviously new to libfuse codebase so i might be misunderstanding some things.